### DISCUSSION TITLE
What is history of a document?

### DISCUSSION THREAD

**Discussion by iamsaumya** [outline/outline#2016] [General]
Will history contain all the revisions that will be made to a document? I want to know when is a revision created for a document. 

Will it be only for these two events? If yes, I don't think a general update to the document creates a revision history. 

https://github.com/outline/outline/blob/b3203857d79f11d7c55c59660692d8337b06e8f2/server/services/revisions.js#L7-L10

PS: I am not sure if this is the best place to ask this question.

**Comment by iamsaumya** [comment]
Hey @tommoor please help me here. What is the expected behaviour of the history feature?

**Reply by tommoor** [reply]
The current expected behaviour is that a revision is created after 5 minutes of inactivity on a document, this is the "debounce" time.

**Comment by ibonelli** [comment]
So this has been moving around, and the functionality now seems to be at [server/queues/processors/RevisionsProcessor.ts](https://github.com/outline/outline/blob/main/server/queues/processors/RevisionsProcessor.ts)

But I get the impression this can be resolved as there is not much more to do within this issue.

I would like though understand how the revisions get created. When I edit a document I see the document changes instead of creating revisions, and I can not find a way to configure what constitutes a revision for me (ie. reduce the debounce time). By the way, I'm hosting Outline myself.

I tested and I do not see the 5 minutes debounce being applied. I always see the main document being updated (I'm checking at DB level as UI only shows document updates --published, moved, latestVersion-- and doesn't show real revisions).
